dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -486,6 +495,57 @@ public void lifeCycleStarted(LifeCycle event) {
+                Session session = hibernate.getSessionFactory().openSession();
+                ManagedSessionContext.bind(session);
+                final Transaction transaction = session.beginTransaction();
+                try {
+                    TokenDAO tokenDAO = new TokenDAO(hibernate.getSessionFactory());
+                    // exclude both gho tokens that the application generates and ghp tokens that we can insert for testing
+                    // https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/
+                    tokenDAO.findAllGitHubTokens().stream().filter(t -> !t.getContent().startsWith("gho_") && !t.getContent().startsWith("ghp_")).forEach(token -> {
+                        try {
+                            // cannot use normal github library
+                            final java.net.http.HttpClient client = java.net.http.HttpClient.newBuilder().build();
+                            HttpRequest request = HttpRequest.newBuilder()
+                                .uri(new URI("https://api.github.com/applications/" + configuration.getGithubClientID() + "/token"))
+                                .header("Accept", "application/vnd.github+json")
+                                .header("Authorization", getBasicAuthenticationHeader(configuration.getGithubClientID(), configuration.getGithubClientSecret()))
+                                .method("PATCH", BodyPublishers.ofString("{\"access_token\":\"" + token.getContent() + "\"}"))
+                                .build();
+                            final HttpResponse<ResetTokenModel> send = client.send(request, new JsonBodyHandler<>(ResetTokenModel.class));
+                            final String token1 = send.body().token;
+                            if (send.statusCode() != HttpStatus.SC_NOT_FOUND) {
+                                token.setContent(token1);
+                                tokenDAO.update(token);
+                                LOG.info("updated token for {}", token.getUsername());
+                            } else {
+                                LOG.error("could not update old style github token for {}, token was not found on github", token.getUsername());
+                            }
+                        } catch (IOException e) {
+                            LOG.error("could not update old style github token for {}", token.getUsername());
+                        } catch (URISyntaxException e) {
+                            LOG.error("could not update old style github token for {} due to syntax issue", token.getUsername());
+                        } catch (InterruptedException e) {
+                            LOG.error("could not update old style github token for {} due to interruption", token.getUsername());
+                            // Restore interrupted state... (sonarcloud suggestion)
+                            Thread.currentThread().interrupt();
```

Comment:
Typically, if we've caught `InterruptedException`, another thread interrupted this thread and wants it to gracefully stop what it's doing and exit: https://stackoverflow.com/questions/2126997/who-is-calling-the-java-thread-interrupt-method-if-im-not
In that case, it might be good to exit the method after this line.

Reply:
Not sure how to break from a foreach, cannot rethrow the exception, but at the same the the IDE claims returning from a foreach is a style-error. Think I'll leave this one since it feels weird to refactor this for an exceptional situation.

Reply:
Yeah, a regular `for` loop might be preferable for big blocks of code that could conceivably throw or return/continue/break.  Perhaps the interrupt scenario isn't so far-fetched...  Suppose we start the application then want to shut it down immediately, and this method has lots of work to do.  The app server will probably try to interrupt this thread and wait a while before outright killing it, so it would probably be faster/cleaner if the thread cooperates.